### PR TITLE
chore: types based on inheritance tune up

### PR DESCRIPTION
**Pull request by egorio**
Small code improvements

### CODE REVIEW

File: packages/codefixes/src/fixes/addMissingTypesBasedOnInheritance/index.ts [hunk 1]
```diff
@@ -39,27 +39,24 @@ export class AddMissingTypesBasedOnInheritanceCodeFix implements CodeFix {
...
-        );
-
-        const ancestorMethodCallSignatures = ancestorMethodType.getCallSignatures();
+        const ancestorMethodCallSignatures = checker
+          .getTypeOfSymbolAtLocation(ancestorMethodSymbol, ancestorMethodSymbol.valueDeclaration)
+          .getCallSignatures();
 
         // Don't support multiple signatures
-        if (ancestorMethodCallSignatures.length > 1) {
+        if (ancestorMethodCallSignatures.length !== 1) {
```
Comment (egorio) [inline packages/codefixes/src/fixes/addMissingTypesBasedOnInheritance/index.ts:47]: Cover cases `== 0` and `> 1`

File: packages/codefixes/src/fixes/addMissingTypesBasedOnInheritance/index.ts [hunk 2]
```diff
@@ -39,27 +39,24 @@ export class AddMissingTypesBasedOnInheritanceCodeFix implements CodeFix {
...
           continue;
         }
 
         const parameterSymbol = ancestorMethodCallSignatures[0].parameters[currentParameterIndex];
 
         // Additional parameter in a child function won't have param in parent function signature
-        if (!parameterSymbol) {
+        if (!parameterSymbol || !parameterSymbol.valueDeclaration) {
```
Comment (egorio) [inline packages/codefixes/src/fixes/addMissingTypesBasedOnInheritance/index.ts:54]: Covers if there is no `valueDeclaration` available for some magical reason

Comment (chadhietala) [reply packages/codefixes/src/fixes/addMissingTypesBasedOnInheritance/index.ts:54]: Any chance you have a test case we can add for this?

Comment (egorio) [reply packages/codefixes/src/fixes/addMissingTypesBasedOnInheritance/index.ts:54]: https://github.com/microsoft/TypeScript/issues/36697 :)

**Review by chadhietala — APPROVED** [review]

**Pull request merged**